docs/csharp/language-reference/operators/boolean-logical-operators.md

```diff
@@ -0,0 +1,159 @@
+
+For the operands of [integral](../keywords/integral-types-table.md) types, the `&`, `|`, and `^` operators perform bitwise logical operations.
+
+## Logical negation operator !
+
+The unary `!` operator computes logical negation of its operand. That is, it produces `true`, if the operand evaluates to `false`, and `false`, if the operand evaluates to `true`:
+
+[!code-csharp-interactive[logical negation](~/samples/snippets/csharp/language-reference/operators/LogicalOperators.cs#Negation)]
+
+## Logical AND operator &amp;
+
+The `&` operator computes the logical AND of its operands. The result of `x & y` is `true` if both `x` and `y` evaluate to `true`. Otherwise, the result is `false`.
+
+The `&` operator evaluates both operands even if the first operand evaluates to `false`, so that the result must be `false` regardless of the value of the second operand. The following example demonstrates that behavior:
+
+[!code-csharp-interactive[logical AND](~/samples/snippets/csharp/language-reference/operators/LogicalOperators.cs#And)]
+
+The [conditional logical AND operator](#conditional-logical-and-operator-) `&&` also computes the logical AND of its operands, but evaluates the second operand only if the first operand evaluates to `true`.
+
+For the operands of integral types, the `&` operator computes [bitwise logical AND](and-operator.md#integer-logical-bitwise-and-operator) of its operands. The unary `&` operator is the [address-of operator](and-operator.md#unary-address-of-operator).
+
+## Logical exclusive OR operator ^
+
+The `^` operator computes the logical exclusive OR, also known as the logical XOR, of its operands. The result of `x ^ y` is `true` if `x` evaluates to `true` and `y` evaluates to `false`, or `x` evaluates to `false` and `y` evaluates to `true`. Otherwise, the result is `false`. That is, for Boolean operands, the `^` operator computes the same result as the [inequality operator](equality-operators.md#inequality-operator-) `!=`.
+
+[!code-csharp-interactive[logical exclusive-OR](~/samples/snippets/csharp/language-reference/operators/LogicalOperators.cs#Xor)]
+
+For the operands of integral types, the `^` operator computes [bitwise logical exclusive OR](xor-operator.md) of its operands.
+
+## Logical OR operator |
+
+The `|` operator computes the logical OR of its operands. The result of `x | y` is `true` if either `x` or `y` evaluates to `true`. Otherwise, the result is `false`.
+
+The `|` operator evaluates both operands even if the first operand evaluates to `true`, so that the result must be `true` regardless of the value of the second operand. The following example demonstrates that behavior:
+
+[!code-csharp-interactive[logical OR](~/samples/snippets/csharp/language-reference/operators/LogicalOperators.cs#Or)]
+
+The [conditional logical OR operator](#conditional-logical-or-operator-) `||` also computes the logical OR of its operands, but evaluates the second operand only if the first operand evaluates to `false`.
```

Comment:
Again:  

> only if the first operand doesn't evaluate to `true`.

Reply:
@BillWagner while your suggestion is correct, I'm afraid readers would translate it to "...evaluates to `false`" anyway ("doesn't evaluate to `true`" might be hard to comprehend), with most proactive of them submitting PR to "fix" wording.

I suggest another fix. I've noticed that I'm not consistent when talking about short-circuiting behavior. Sometimes I mention condition for short-circuiting ("...the first operand evaluates to `true`" in case of `||`), in other cases — condition for evaluation of the second operand ("...the first operand doesn't evaluate to `true`"). I think only one condition should be mentioned. I suggest to describe condition for short-circuit; in particular, because the [spec](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#user-defined-conditional-logical-operators) definition of user-defined conditional operators says when to short-circuit.

Then, the paragraph in question would read:
>The conditional logical OR operator `||` also computes the logical OR of its operands, but doesn't evaluate the second operand if the first operand evaluates to `true`.

Reply:
Resolved again.